doc/reference/vm_live_migration_internals.md

```diff
@@ -0,0 +1,127 @@
+---
+discourse: ubuntu:[Online&#32;VM&#32;live-migration&#32;(QEMU&#32;to&#32;QEMU)](50734)
+---
+
+(vm-live-migration-internals)=
+# LXD VM live migration implementation
+
+LXD supports {ref}`live-migration-vms` of virtual machines by streaming instance state from a source {abbr}`QEMU (Quick Emulator)` to a target QEMU. VM live migration is supported for all storage pool types.
```

Comment:
The `{ref}` expands the header target, which is "Live migration for virtual machines". So in the RTD preview, this line currently reads: "LXD supports Live migration for virtual machines of virtual machines..."
Just need to remove the text "of virtual machines" here.

Or if you prefer to keep the "of" phrasing, use this syntax to override fetching the link text from the reference with your own link text:
```
{ref}`your preferred link text <header_target>`
```

As in:
```
{ref}`live migration of virtual machines <live-migration-vms>` by streaming ...
```

Reply:
Good to know, thanks! The `{ref}` expansion maintains the original casing for the header target. For now I'll just reword the sentence to start with "Live migration for virtual machines", but we may want to consider defining a Sphinx directive called something like `{ref_lowercase}` which converts the casing the lower.